gemfirexd/core/src/main/java/com/pivotal/gemfirexd/internal/engine/sql/execute/MemberStatisticsMessage.java

```diff
@@ -179,9 +186,29 @@ private String getUserDir(){
     return System.getProperty("user.dir");
   }
 
+  private String getLogFile() {
+    Logger rootLogger = Logger.getRootLogger();
+    Appender appender;
+    if (rootLogger != null) {
+      Enumeration<?> e = rootLogger.getAllAppenders();
+      while (e.hasMoreElements()) {
+        appender = (Appender)e.nextElement();
+        if (appender instanceof FileAppender) {
+          return ((FileAppender)appender).getFile();
+        }
+      }
+    }
+    return "";
+  }
+
   private String getProcessId(){
-    String[] processDetails = ManagementFactory.getRuntimeMXBean().getName().split("@");
-    return processDetails[0];
+    String processId = "";
+    try {
+      processId = Integer.toString(ProcessUtils.identifyPid());
+    } catch (PidUnavailableException e) {
+      logger.warn("Process ID NOT AVAILABLE..", e);
+    }
```

Comment:
@snappy-sachin Use NativeCalls.getInstance().getProcessId(). That works for all known platforms using native JNA and is much more reliable that this custom parsing so should be preferred everywhere. @rishitesh